rclpy/rclpy/executors.py

```diff
@@ -762,7 +762,11 @@ def _spin_once_impl(
         except ConditionReachedException:
             pass
         else:
-            self._executor.submit(handler)
+            def handler_wrapper(handler):
+                handler()
+                if handler.exception() is not None:
+                    raise handler.exception()
```

Comment:
I'm not sure if it's a great idea to throw an exception from the executor when any of the callbacks fail.
It would make everything fail when only one callback failed.

Maybe something else like logging an error is a good idea, I'm not sure.

Reply:
That does bring up a good point. The main issue I had in https://github.com/ros2/rclpy/issues/983 is silently failing, and the message that no retrieval of exception could be very delayed leading to head scratching.

If we decide to not to throw exception from the executor, would logging it out be better, assuming that would be immediate? As a preventative measure all the timer call backs in our code are wrapped in a `try except` block but was a odd experience figuring that out from a user perspective.

Reply:
> If we decide to not to throw exception from the executor, would logging it out be better, assuming that would be immediate? 

Yes, logging is better than nothing.
And if you log from here that should be immediate.

Reply:
+1 to logging. Logging and ignoring it has some precedent [in concurrent.futures](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.add_done_callback).

An enhancement might be to allow the user to set a custom exception handler, as [`asyncio` seems to allow people to do](https://docs.python.org/3/library/asyncio-eventloop.html#error-handling-api).

Reply:
addressed in https://github.com/ros2/rclpy/pull/984/commits/cda6aeeccb78777101235fafcd4ca6a14acc70d9